assets/diagrams/promote.mmd

```diff
@@ -0,0 +1,13 @@
+graph LR
+A(npm dist-tag add)
+B(fa:fa-server warehouse.ai)
+C{fa:fa-database No-SQL Database}
+D{fa:fa-database npm-like registry}
+E>BUILD]
+
+B -.-> E
+B --> |"update module {version@env}"| C
+D --> |"fetch @version tarball"| B
```

Comment:
should the communication with `npm-like registry` here be to write the dist-tags.  IIRC it doesn't grab the tarball here, just sends along the dist-tag and queues some builds for the version/environment

Reply:
it needs the tarball to build though, I'll add that write of `dist-tags`

Reply:
ah, you're right, I missed that part, I thought carpenterd just did that as part of the build process, didn't see that carpenterd requires that the full tarball be sent everytime.